src/lib.rs

```diff
@@ -10,252 +10,160 @@ extern crate blkid_sys;
-                1 => Ok(true),
-                _ => Err(BlkidError::new(format!(
-                    "Unknown return code from \
-                     blkid_probe_has_value: {}",
-                    ret_code
-                ))),
-            }
+        match ret_code {
+            0 => Ok(false),
+            1 => Ok(true),
+            _ => Err(BlkidError::LibBlkidHasValue { ret_code }),
         }
     }
-    /// number of values in probing result
+
+    /// Number of values in probing result
     pub fn numof_values(&self) -> Result<i32, BlkidError> {
-        unsafe {
-            let ret_code = blkid_probe_numof_values(self.probe);
-            if ret_code < 0 {
-                return Err(BlkidError::new(get_error()));
-            }
-            Ok(ret_code)
+        let ret_code = unsafe { blkid_probe_numof_values(self.probe) };
+        if ret_code < 0 {
+            return Err(BlkidError::get_error());
         }
+
+        Ok(ret_code)
     }
 
-    /// Retrieve the Nth item (Name, Value) in the probing result, (0..self.numof_values())
+    /// Retrieve the Nth item `(Name, Value)` in the probing result, (0..self.numof_values())
     pub fn get_value(&self, num: i32) -> Result<(String, String), BlkidError> {
         let mut name_ptr: *const ::libc::c_char = ptr::null();
         let mut data_ptr: *const ::libc::c_char = ptr::null();
         let mut len = 0;
 
-        unsafe {
-            let ret_code =
-                blkid_probe_get_value(self.probe, num, &mut name_ptr, &mut data_ptr, &mut len);
-            if ret_code < 0 {
-                return Err(BlkidError::new(get_error()));
-            }
-            let name_value = CStr::from_ptr(name_ptr as *const ::libc::c_char);
-            let data_value = CStr::from_ptr(data_ptr as *const ::libc::c_char);
-            Ok((
-                name_value.to_string_lossy().into_owned(),
-                data_value.to_string_lossy().into_owned(),
-            ))
+        let ret_code = unsafe {
+            blkid_probe_get_value(self.probe, num, &mut name_ptr, &mut data_ptr, &mut len)
+        };
+        if ret_code < 0 {
+            return Err(BlkidError::get_error());
         }
+
+        let name_value = unsafe { CStr::from_ptr(name_ptr as *const ::libc::c_char) };
+        let data_value = unsafe { CStr::from_ptr(data_ptr as *const ::libc::c_char) };
+        Ok((
+            name_value.to_string_lossy().into_owned(),
+            data_value.to_string_lossy().into_owned(),
+        ))
     }
 
     // https://github.com/karelzak/util-linux/blob/master/Documentation/blkid.txt
     /// Retrieve the value of a specific attribute for a particular device.  
     /// This can be used to determine attributes such as TYPE, UUID, LABEL, and PARTUUID
-    /// Returns empty string if the requested attribute is not set for a particular device
+    ///
+    /// # Returns
+    /// Empty string if the requested attribute is not set for a particular device
```

Comment:
I know this isn't related to your PR but I wonder if this should return an Option instead of an empty string. That would be more rust like